pulsar-broker/src/test/java/org/apache/pulsar/client/api/DispatcherBlockConsumerTest.java

```diff
@@ -998,6 +999,11 @@ public void testBrokerDispatchBlockAndSubAckBackRequiredMsgs() {
              * maxUnAckPerDispatcher limit
              **/
             consumer1Sub2 = (ConsumerImpl<byte[]>) pulsarClient.newConsumer().topic(topicName)
+                    // in the scope of this test is important to not group acks because
+                    // the ack for messages happens after receiving a new message from the topic.
+                    // If the acks are sent in batch it may happen that the dispatcher is still blocked until
+                    // it receives the batched acks and the Consumer#receive will return null.
+                    .acknowledgmentGroupTime(0, TimeUnit.SECONDS)
```

Comment:
Using ackReceiptEnable can help as well.
Because you cam wait for the server to process the ack

Reply:
@eolivelli `ackReceiptEnable` would also work but it slows down the test a lot (5 secs vs 1 min) so I'd prefer to disable the ack grouping. For the scope of the test this is irrelevant